core/src/jsMain/kotlin/dev/fritz2/binding/store.kt

```diff
@@ -258,16 +260,86 @@ open class RootStore<T>(
     /**
      * a simple [SimpleHandler] that just takes the given action-value as the new value for the [Store].
      */
-    override val update = this.handle<T> { _, newValue -> newValue }
+    override val update = this.handle<D> { _, newValue -> newValue }
 }
 
 /**
- * convenience method to create a simple [RootStore] without any handlers, etc.
+ * A [ValidatingStore] is a [RootStore] which also contains a [Validation] for its model.
  *
- * @param initialData the first current value of this [Store]
- * @param id the id of this store. ids of [SubStore]s will be concatenated.
+ * @param initialData first current value of this [Store]
+ * @param validation [Validation] function to use at the data on this [Store].
+ * @param validateAfterUpdate flag to decide if a new value gets automatically validated after setting it to the [Store].
+ * @param id id of this [Store]. Ids of [SubStore]s will be concatenated.
  */
-fun <T> storeOf(initialData: T, id: String = Id.next()) = RootStore(initialData, id)
+open class ValidatingStore<D, T, M>(
+    initialData: D,
+    private val validation: Validation<D, T, M>,
+    val validateAfterUpdate: Boolean = true,
+    override val id: String = Id.next()
+) : RootStore<D>(initialData, id) {
+
+    private val validationMessages: MutableStateFlow<List<M>> = MutableStateFlow(emptyList())
+
+    /**
+     * [Flow] of the [List] of validation-messages.
+     * Use this [Flow] to render out the validation-messages.
+     */
+    val messages: Flow<List<M>> = validationMessages.asStateFlow()
+
+    /**
+     * Resets the validation result.
+     *
```

Comment:
Beware that cleaning the messages should not be done, if the [data] [Flow] remains in an invalid state. Please refer to the class's description for details about the need for a sound data and messages state.